src/component/tab-menu/TabMenu.ts

```diff
@@ -0,0 +1,50 @@
+import { MenuProps } from "../../types/menu.ts";
+
+export default class TabMenu {
+  container: HTMLElement;
+  #currentMenu: MenuProps = "all";
+  onTabChange: () => void = () => {};
+
+  constructor() {
+    this.container = document.createElement("div");
+    this.container.classList.add("tabmenu-container");
+    this.render();
+  }
+
+  render() {
+    this.container.innerHTML = `
+     <button class="tabmenu-item" data-tab="all">모든 음식점</button>
+     <button class="tabmenu-item" data-tab="favorite">자주 가는 음식점</button>
+    `;
+    this.setActiveTabStyle();
+    this.handleCurrentMenu();
+  }
+
+  handleCurrentMenu() {
+    this.container.querySelectorAll(".tabmenu-item")?.forEach((tabMenu) =>
+      tabMenu.addEventListener("click", (event: Event) => {
+        const target = event.target as HTMLButtonElement | null;
+        if (!target) return;
```

Comment:
as를 통한 단언은 컴파일러보다 개발자가 더 많은 정보를 알고 있을 때 사용한다고 생각하면 좋습니다.
그런 면에서 위와 같은 상황은 as를 사용하기 최적의 상황은 아닌 것 같아요. 
물론 지금은 tabmenu-item DOM이 버튼이라는것을 확신 할 수 있지만, 
조금 더 방어적으로 instanceof 타입 가드를 사용해볼 수 있을 것 같습니다.

```suggestion
 if (!(event.target instanceof HTMLButtonElement)) return;
  const target = event.target;
```

타입가드는 다양한 상황에 많이 사용하게 될테니 따로 학습해보시는것을 추천드립니다.

Reply:
위 내용도 코드 전반적으로 사용한 모든 곳에 적용해봅시다.